core/common/assets/js/utils/environment.js

```diff
@@ -1,14 +1,49 @@
-const userAgent = navigator.userAgent,
+const matchUserAgent = ( UserAgentStr ) => {
+		return userAgent.indexOf( UserAgentStr ) >= 0;
+	},
+
+	userAgent = navigator.userAgent,
+
+	// Solution influenced by https://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser
+
+	// Opera 8.0+
+	isOpera = ( !! window.opr && !! opr.addons ) || !! window.opera || matchUserAgent( ' OPR/' ),
+
+	// Firefox 1.0+
+	isFirefox = typeof InstallTrigger !== 'undefined' && matchUserAgent( 'Firefox' ),
+
+	// Safari 3.0+ "[object HTMLElementConstructor]"
+	isSafari = /^((?!chrome|android).)*safari/i.test( userAgent ) || /constructor/i.test( window.HTMLElement ) ||
+		( ( p ) => {
+			return '[object SafariRemoteNotification]' === p.toString();
+		} )( ! window.safari || ( typeof safari !== 'undefined' && safari.pushNotification ) ),
+
+	// Internet Explorer 6-11
+	isIE = /Trident|MSIE/.test( userAgent ) && ( /*@cc_on!@*/false || !! document.documentMode ),
+
+	// Edge 20+
+	isEdge = ( ! isIE && !! window.StyleMedia ) || matchUserAgent( 'Edg' ),
+
+	// Chrome 1+
+	isChrome = !! window.chrome && matchUserAgent( 'Chrome' ),
+
+	// Blink engine
+	isBlink = matchUserAgent( 'Chrome' ) && !! window.CSS,
+
+	// Apple Webkit engine
+	isAppleWebkit = matchUserAgent( 'AppleWebKit' ) && ! isBlink,
+
 	environment = {
-		webkit: -1 !== userAgent.indexOf( 'AppleWebKit' ),
-		firefox: -1 !== userAgent.indexOf( 'Firefox' ),
-		ie: /Trident|MSIE/.test( userAgent ),
-		edge: -1 !== userAgent.indexOf( 'Edg' ),
-		mac: -1 !== userAgent.indexOf( 'Macintosh' ),
-		safari: /^((?!chrome|android).)*safari/i.test( userAgent ),
-		opera: -1 !== userAgent.indexOf( 'OPR' ),
+		appleWebkit: isAppleWebkit,
+		blink: isBlink,
+		chrome: isChrome,
+		edge: isEdge,
+		firefox: isFirefox,
+		ie: isIE,
+		mac: matchUserAgent( 'Macintosh' ),
+		opera: isOpera,
+		safari: isSafari,
+		webkit: matchUserAgent( 'AppleWebKit' ),
 	};
 
-environment.chrome = -1 !== userAgent.indexOf( 'Chrome' ) && ! environment.opera && ! environment.edge;
```

Comment:
Why are the additional checks removed?
 ` ! environment.opera && ! environment.edge;`

Reply:
Good point.
We need to discuss this.